File: backend/hotelEase/hotelEase.Services/Database/Payment.cs
using System;
using System.Collections.Generic;

namespace hotelEase.Services.Database;

public partial class Payment
{
    public int Id { get; set; }

    public int ReservationId { get; set; }

    public string Provider { get; set; } = null!;

    public string ProviderPaymentId { get; set; } = null!;

    public decimal Amount { get; set; }

    public string Currency { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool? IsDeleted { get; set; }

    public DateTime? DeletedTime { get; set; }

    public virtual Reservation Reservation { get; set; } = null!;
}

File: UI/hotelease_desktop/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int? id;
  final int? reservationId;
  final String? provider;
  final String? providerPaymentId;
  final double? amount;
  final String? currency;
  final String? status;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Payment({
    required this.id,
    required this.reservationId,
    required this.provider,
    required this.providerPaymentId,
    required this.amount,
    required this.currency,
    required this.status,
    required this.createdAt,
    this.updatedAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
